#! /bin/sh -e

# DP: * java/lang/Class.java (internalGetFields): Use LinkedHashSet.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

2007-08-15  Tom Tromey  <tromey@redhat.com>

	* java/lang/Class.java (internalGetFields): Use LinkedHashSet.
	* classpath/lib/java/lang/Class.class: Rebuilt.

Index: libjava/java/lang/Class.java
===================================================================
--- libjava/java/lang/Class.java	(revision 127517)
+++ libjava/java/lang/Class.java	(revision 127518)
@@ -56,7 +56,7 @@
 import java.security.ProtectionDomain;
 import java.util.ArrayList;
 import java.util.Arrays;
-import java.util.HashSet;
+import java.util.LinkedHashSet;
 import java.util.HashMap;
 import java.util.Collection;
 import java.lang.reflect.AnnotatedElement;
@@ -473,7 +473,7 @@
    */
   private Field[] internalGetFields()
   {
-    HashSet set = new HashSet();
+    LinkedHashSet set = new LinkedHashSet();
     set.addAll(Arrays.asList(getDeclaredFields(true)));
     Class[] interfaces = getInterfaces();
     for (int i = 0; i < interfaces.length; i++)
